include/mapbox/vector_tile/attribute_handler.hpp

```diff
@@ -0,0 +1,184 @@
+#pragma once
+
+#include <mapbox/feature.hpp>
+#include <vtzero/vector_tile.hpp>
+#include <memory>
+
+namespace mapbox {
+namespace vector_tile {
+namespace detail {
+
+struct AttributeHandler {
+
+    using map_type = mapbox::feature::property_map;
+    using vec_type = std::vector<mapbox::feature::value>;
+
+    struct handler_store {
+        std::unique_ptr<map_type> map;
+        std::unique_ptr<vec_type> vec;
+        vtzero::data_view last_key;
+        vtzero::scaling const* scaling;
+        bool is_map;
+    };
+
+    std::vector<handler_store> stack;
+    vtzero::layer const& layer;
```

Comment:
@joto I had to carry the `vtzero::layer` into my AttributeHandler to support scaling of number lists. This makes it odd because I normally only was required to have `vtzero::feature` at this point. Is this expected? Should there be a method to expose this on the feature object?

Reply:
The `feature` contains a pointer to the `layer`, so the `feature` should be enough. I'll look into it.

Reply:
I added an accessor function `feature::layer()` to get to the layer now. This should make things a bit easier. We might add some functions to magically decode the scaling for us, but I am not sure exactly how this should look like.